doc/src/manual/command-line-interface.md

```diff
@@ -46,8 +46,29 @@ but at the conclusion of executing a script or expression, `julia` will attempt
 `Main.main(Base.ARGS)` if such a function `Main.main` has been defined and this behavior was opted into
 by using the `@main` macro.
 
-This feature is intended to aid in the unification
-of compiled and interactive workflows. In compiled workflows, loading the code that defines the `main`
+To see this feature in action, consider the following definition:
+```jldoctest
+function (@main)(ARGS)
+    open("out.log", "w") do io
+        println(io, "My pet is $(ARGS[1])")
+    end
+    return nothing
+end
```

Comment:
`(@main)(args) = println("Hello World!")` was a better example in most ways and this is better in that it uses args. How about `(@main)(args) = println("Hello $(args[1])!")`?